### ISSUE TITLE
DWCA reader incompatibility with Python v3.5

### ISSUE THREAD

**Issue by pedrosiracusa** [BelgianBiodiversityPlatform/python-dwca-reader#62]
The following OS error is thrown when instancing DwCAReader for Python v3.5 on Windows 7:
`dwca = DwCAReader('results.zip')` 

`OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: ':'`

I found the problem resides on file read.py line 273
`tmp_dir = mktemp()[1]`
Apparently by indexing `mktemp()` we are setting `tmp_dir` to ':', the second character on the path for windows OS. 

As explained on python docs the function `mktemp()` returns the absolute path for the new temporary directory. By making `tmp_dir = mktemp()` I was able to overcome the issue.

**Comment by silenius** [comment]
the `[1]` looks strange... `mkdtemp` returns a string on both Python2 and 3:

```
$ python3 -c "from tempfile import mkdtemp; print(mkdtemp())" 
/tmp/tmp59jo5i4e
$ python2 -c "from tempfile import mkdtemp; print(mkdtemp())"
/tmp/tmprnsllv
```

**Comment by niconoe** [comment]
Thanks @pedrosiracusa, you may be the first brave person to try python-dwca-reader under Windows!

I plan to investigate further a bit later, when I have a Windows box around (it does indeed work with Python 3 under Linux and Mac OS).

If I can make it work relatively easily under Windows too, I hope to release a new version that fix this next week.

Thanks again for your report !

**Comment by niconoe** [comment]
Seems `mkdtemp()` was misused from a long time, and Windows reveal the bug... Working on a fix!

**Comment by niconoe** [comment]
@pedrosiracusa, could you try again with the freshly released v0.10.1?

Thanks!

**Comment by pedrosiracusa** [comment]
Thanks for the fix @niconoe ! `mkdtemp()` is now working properly, although there's still an issue with the closing method `dwca.close()` for (Windows7+Spyder) setup. 
Thanks!
